File: Server_trunk/res/scripts/cell/Spell/CastConditions/ConditionBase.py
# -*- coding: utf-8 -*-

from KBEDebug import *

from ..SpellDef import SpellStatus

class ConditionBase(object):
	"""
	1.判断施法者或受术者自身是否满足条件
	2.判断两者之间的关系是否满足条件
	"""
	def init( self, dataSection ):
		"""
		@param DataSection: instance of PyDataSection
		@return: None
		"""
		pass

	def verify( self, target, otherTarget ):
		"""
		判断施法者或受术者自身是否满足条件
		@param target: Entity; 施法者或受术者
		@param otherTarget: Entity; 与target判断关系的目标对象（若只判断自身是否满足条件，该参数可以为None）
		"""
		return SpellStatus.OK;

File: Client_trunk2/Assets/Scripts/Spells/CastConditions/ConditionBase.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

namespace SPELL
{
    public abstract class ConditionBase
    {
        /// <summary>
        /// 类型映射表
        /// 注意：此表需要与服务器端的保持一致
        /// </summary>
        private static Dictionary<int, System.Type> s_conditionMap = new Dictionary<int, System.Type>()
        {
            //{1, typeof(NotEntityActionFlags)},
            {4, typeof(EffectStatusCondition)}
        };

        public static ConditionBase CreateCondition(int type)
        {
            if (!s_conditionMap.ContainsKey(type))
                return null;

            var cond = (ConditionBase)System.Activator.CreateInstance(s_conditionMap[type]);
            return cond;
        }

        public virtual void Init(DataSection.DataSection dataSection)
        {

        }

        /// <summary>
        /// 判断施法者或受术者自身是否满足条件
        /// </summary>
        /// <param name="src">caster or receiver</param>
        public virtual SpellStatus Verify(AvatarComponent src, AvatarComponent dst)
        {
            return SpellStatus.OK;
        }
    }
}
